### PR TITLE
Build failure fix for latest LLVM

### PR DESCRIPTION
**Pull request by saranyakrish14**
Summary : LLVM>=11 no longer supports implicit conversion from llvm::StringRef-> std::string . Because of this glow contains lot of broken places and does not compile anymore. Made a fix on such places by explicitly converting llvm::StringRef->std::string.

Along with that when trying to set the alignment of global variable using setAlignment method ,it is reported as "Deprecated" in LLVM 10 and throws error in LLVM>=11 .Fixed that issue using llvm::MaybeAlign.

Related issue:#5068

### CODE REVIEW

File: include/glow/LLVMIRCodeGen/LLVMBackend.h [hunk 1]
```diff
@@ -59,21 +59,21 @@ class LLVMBackendOptions {
 public:
   LLVMBackendOptions();
   /// \returns target triple used by this backend.
-  llvm::StringRef getTarget() const { return target_; }
+  std::string getTarget() const { return target_; }
```
Comment (opti-mix) [inline include/glow/LLVMIRCodeGen/LLVMBackend.h:62]: In cases, where the result is not supposed to be modified, it is better to return `const std::string &`

**Review by opti-mix — COMMENTED** [review]
@saranyakrish14 Thanks for working on this!

**Comment by opti-mix** [comment]
@jfix71 Do you think we can add a CI which would use >= LLVM 11 to build Glow to detect any future uses of deprecated APIs? Without it the probability of introducing the same issues again is very high.

File: include/glow/LLVMIRCodeGen/LLVMBackend.h (hunk 1 above)
Comment (saranyakrish14) [reply include/glow/LLVMIRCodeGen/LLVMBackend.h:62]: ok, I will make the changes.

**Comment by opti-mix** [comment]
@saranyakrish14 Thanks for addressing the review comments!

I'm trying to understand the scope of this PR. The changes seem to include not only StringRef related fixes and other LLVM11 related fixes, but also some test names fixes, deferred weights error fixes, etc. Is it intentional or is it because you merged the upstream changes? It would be good to have separate PRs/commits for different kinds of fixes to keep things clean.

**Comment by saranyakrish14** [comment]
@opti-mix  Its because of merging the upstream changes. My bad. I have reverted back to my last commit .

**Comment by jfix71** [comment]
>Its because of merging the upstream changes. My bad. I have reverted back to my last commit .

@saranyakrish14 CI has some weird errors here -- I am wondering if it is because of a bad merge from upstream master. Can you try merging to latest Glow master and push up the update here, so we can see if it resolves the CI issue? If it doesn't help we will investigate further.

**Comment by opti-mix** [comment]
@saranyakrish14 Something is wrong with this PR in terms of how it triggers CI. Other PRs trigger the OSS CI using PyTorch/Glow as the project/org (see e.g. #5643), but your PR somehow triggers Circle CI using your own username, instead of the usual PyTorch/Glow as the project/org.

Do you have any ideas why it does so? Maybe you have your own CircleCI account it somehow used when you submit your PR? If so, try to sign-out from CircleCI when you e.g. update this PR or create a PR.

One other thing to try is to create another PR based on the same changes and see if it will be any better.

**Pull request closed without merging**